sigma/plugins/roles/rmselfcolorrole.py

```diff
@@ -0,0 +1,58 @@
+import discord
+from sigma.core.permission import check_man_roles
+from sigma.core.rolecheck import matching_role
+
+async def rmselfcolorrole(cmd, message, args):
+    # If no args, send help message.
+    if not args:
+        await message.channel.send(cmd.help())
+        return
+
+    # checks permissions
+    if not check_man_roles(message.author, message.channel):
+        out_content = discord.Embed(type='rich', color=0xDB0000, title='⛔ Insufficient Permissions. Server Admin Only.')
+        await message.channel.send(None, embed=out_content)
+        return
+
+    # find role to delete
+    name = ' '.join(args)
+    target_role = matching_role(message.guild, name)
+
+    # check if role exists
+    if not target_role:
+        out_content = discord.Embed(type='rich', color=0xFF9900, title='❗ Error')
+        out_content.add_field(name='Role Not Found', value='I was unable to find **' + name + '** on this server.')
+        await message.channel.send(None, embed=out_content)
+        return
+
+    # check if role is color role
+    if target_role.colour.value == discord.Colour.default():
+        out_content = discord.Embed(type='rich', color=0xFF9900, title='❗ Error')
+        out_content.add_field(name='Not a Color Role', value='Role **' + name + '** is not a color role. Use `destroyrole` instead.')
+        await message.channel.send(None, embed=out_content)
+        return
+
+    # check if role is toggleable
+    # retrieve list of toggleable roles
+    try:
+        self_roles = cmd.db.get_settings(str(message.guild.id), 'SelfRoles')
+    except:
+        out_content = discord.Embed(type='rich', color=0xFF9900, title='❗ Error')
+        out_content.add_field(name='Could Not Retrieve List of Self Roles', value='Please contact the bot owner.')
+        await message.channel.send(None, embed=out_content)
+        return
+
+    # check
+    for self_role in self_roles:
+        if self_role == target_role.id or self_role == target_role.name:
+            break
+    else:
+        out_content = discord.Embed(type='rich', color=0xFF9900, title='❗ Error')
+        out_content.add_field(name='Not a Color Role', value='Role **' + name + '** is not a self role. Use `destroyrole` instead.')
```

Comment:
I recommend using
```
f'Role **{name}** is not a self role. Use `destroyrole` instead.'
```
instead of
```
'Role **' + name + '** is not a self role. Use `destroyrole` instead.'
```
This goes for all other strings